src/main/scala/io/iohk/ethereum/db/storage/BlockHeadersStorage.scala

```diff
@@ -24,7 +25,23 @@ class BlockHeadersStorage(val dataSource: DataSource) extends KeyValueStorage[Bl
   override def valueDeserializer: (IndexedSeq[Byte]) => BlockHeader =
     (encodedBlockHeader: IndexedSeq[Byte]) => rlpDecode[BlockHeader](encodedBlockHeader.toArray)
 
-  override protected def apply(dataSource: DataSource): BlockHeadersStorage = new BlockHeadersStorage(dataSource)
+  override protected def apply(dataSource: DataSource): BlockHeadersStorage =
+    new BlockHeadersStorage(dataSource, blockHeadersNumbersStorage)
+
+  def get(blockNumber: BigInt): Option[BlockHeader] = if (blockNumber == 0) {
+    //todo get block from config, or we will put 0 block to storage at app startup if it is not there?
+    None
+  } else {
+    blockHeadersNumbersStorage.get(blockNumber).flatMap(key => super.get(key))
```

Comment:
I need to check this again, but take a loot at this EthereumJ method: https://github.com/ethereum/ethereumj/blob/deb6f976392b2b5a7c6229670b07275495683c89/ethereumj-core/src/main/java/org/ethereum/db/IndexedBlockStore.java#L149

You can get to it if you navigate starting from https://github.com/ethereum/ethereumj/blob/deb6f976392b2b5a7c6229670b07275495683c89/ethereumj-core/src/main/java/org/ethereum/jsonrpc/JsonRpcImpl.java#L712